libtenzir/src/execution_node.cpp

```diff
@@ -348,8 +348,8 @@ struct exec_node_state {
       }
       if (not std::holds_alternative<generator<Output>>(*output_generator)) {
         return caf::make_error(
-          ec::logic_error, fmt::format("{} expected {}, but got {}", *self,
-                                       operator_type_name<Output>(),
+          ec::logic_error, fmt::format("{} {} expected {}, but got {}", *self,
+                                       op->name(), operator_type_name<Output>(),
```

Comment:
This error message would need to be reformatted. "exec-node internal-save-without-progress failed to start" is not very user-friendly.

Reply:
This error is not only shown on internal logic errors, but also when the user writes an invalid pipeline, as is evident by the integration test reference files:

```
> from file:///foo.json read json

error: the file `/foo.json` does not exist
 --> <input>:1:13
  |
1 | from file:///foo.json read json
  |             ^^^^^^^^^ 
  |
error: pipeline-executor aborts start because an execution node failed to start
 = note: exec-node internal-save-without-progress failed to start
 = note: exec-node load failed to instantiate operator
 = note: could not instantiate loader
 = note: pipeline failed to start
```

Reply:
This is a fallout from the recent error propagation changes. We should make it so that the second error is not shown at all.

Reply:
This problem only exists for loaders and savers right now as they have a separate error path still. I'm not sure what to do about that.